### PR TITLE
A simpler ActorRefProvider config #20649

### PR DESCRIPTION
**Pull request by johanandren**
Refs #20649

### CODE REVIEW

File: akka-actor/src/main/resources/reference.conf [hunk 1]
```diff
@@ -83,9 +83,11 @@ akka {
 
   actor {
 
+    # Either one of "local", "remote" or "cluster" or the
     # FQCN of the ActorRefProvider to be used; the below is the built-in default,
-    # another one is akka.remote.RemoteActorRefProvider in the akka-remote bundle.
-    provider = "akka.actor.LocalActorRefProvider"
+    # note that "remote" and "cluster" requires the akka-remote and akka-cluster
+    # artifacts to be on the classpath.
+    provider = "local"
```
Comment (ktoso) [inline akka-actor/src/main/resources/reference.conf]: awesome :)

Comment (johanandren) [inline akka-actor/src/main/resources/reference.conf]: It came to me as a daydream while teaching AAS

**Comment by ktoso** [comment]
LGTM, very nice improvement 👍

**Comment by drewhk** [comment]
LGTM, Great!

File: akka-actor/src/main/scala/akka/actor/ActorSystem.scala [hunk 2]
```diff
@@ -166,7 +166,16 @@ object ActorSystem {
...
-    final val ProviderClass: String = getString("akka.actor.provider")
+    final val ProviderClass: String = {
+      val shortNames = Map(
+        "local" -> classOf[LocalActorRefProvider].getName,
+        // these two cannot be referenced by class as they may not be on the classpath
+        "remote" -> "akka.remote.RemoteActorRefProvider",
+        "cluster" -> "akka.cluster.ClusterActorRefProvider"
+      )
+      val value = getString("akka.actor.provider")
+      shortNames.getOrElse(value, value)
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/actor/ActorSystem.scala]: I'd write this as 
```
getString("akka.actor.provider") match {
  case "local" => ...
  case "remote" => ...
  case fqcn => fqcn
}
```

File: akka-actor/src/main/resources/reference.conf (hunk 1 above)
Comment (hepin1989) [inline akka-actor/src/main/resources/reference.conf]: Nice movement:)

**Comment by patriknw** [comment]
LGTM

**Pull request merged**